src/haywire/route_compare_method.c

```diff
@@ -21,7 +21,7 @@ int hw_route_compare_method(char *url, char* route)
     char prefix;
     int match = 0;
     
-    strcpy(route_key, route);
+    strncpy(route_key, route, 2048);
```

Comment:
I'd do this with sizeof, and you need to null-terminate the string in the case it's longer than 2047 bytes.

  strncpy(route_key, route, sizeof(route_key));
  route_key[sizeof(route_key)-1] = '\n';

e.g. 
https://en.wikipedia.org/wiki/Sizeof#Using_sizeof_with_arrays